tsconfig.json

```diff
@@ -15,7 +15,8 @@
     "noEmit": true,
     "jsx": "preserve",
     "typeRoots": ["./node_modules/@types"],
-    "baseUrl": "src"
+    "baseUrl": "src",
+    "downlevelIteration": true
```

Comment:
What are the implications of this configuration change?

Reply:
One of the modules uses a custom class that someone wrote that allows you to call it's members like an array. It's the EventGrouper class. I didn't 100% get why it needed to be like that, but this change is necessary for typescript to allow iteration in that class.

Reply:
I'd also 100% be down to axe that class, disc is the only spec that uses it as far as I can tell.

Reply:
I think I authored it back when it was looking like the only sane way of handling penance and Atonement was to group events within a threshold, but from the looks of things `downLevelIteration` brings typescript closer inline as a superset of ES so I'm not certain here. I'll keep this in mind as I'm working through the Discipline stuff for shadowlands and see how it pans out.

Reply:
Was this resolved? I wasn't sure if the final decision was to actually change tsconfig or to add some tsignores where the different iterables were used

Reply:
I think the options are

1) Change the config
2) Rewrite the module

I don't think that this is something that will ts-ignore well.

Reply:
@rp4rk From what you looked into is this config safe to merge in?